billing/credit/service.go

```diff
@@ -32,15 +32,21 @@ func (s Service) Add(ctx context.Context, cred Credit) error {
 		return fmt.Errorf("credit amount is negative")
 	}
 	// check if already credited
-	if t, err := s.transactionRepository.GetByID(ctx, cred.ID); err == nil && t.ID != "" {
+	t, err := s.transactionRepository.GetByID(ctx, cred.ID)
+
+	if err != nil && err != ErrNotFound {
```

Comment:
better we use `errors.Is` instead of != check to make sure it unwraps correctly?

Reply:
Done